pkg/cli/rm_helper.go

```diff
@@ -23,7 +23,7 @@ func getSecretsToRemove(arg string, client client.Client, cmd *cobra.Command) ([
 	}
 
 	for _, secret := range secrets {
-		if strings.HasPrefix(secret.Name, arg+".") {
+		if after, found := strings.CutPrefix(secret.Name, arg+"."); found && !strings.ContainsRune(after, '.') {
```

Comment:
This makes sense to me, but I think you should add a comment above it to explain the purpose of this logic, because I could see myself finding it in the future and getting confused.

Reply:
Added a comment